docs/swagger2.yml

```diff
@@ -0,0 +1,1086 @@
+swagger: '2.0'
+
+# TODO(?): pagination should only list resource names or the entire resource?
+# TODO(?): triggers and functions OR trigs and funcs ?
+# TODO(?): version change? /v2 ?
+# TODO(?): remove stats from call?
+# TODO(?): amend log api? if we have /app/logs we could tail logs on an app (or trig?) and have point lookup for call still
+# TODO(?): 'namespace' or 'ns_id' field in objects? 'app_id' or 'app_name' in objects (trigger, app, etc)?
+#           really need to figure this out, e.g. app right now has a weird mix of id and name. we should pick one.
+# TODO(?): annotations & config & event.extensions redundant to a degree? naming confusing? discuss.
+# TODO(?): add /run API to swagger doc? (only covers API module atm)
+# TODO(?): should trigger have an 'enabled' field to easily toggle on and off via API?
+# TODO: let's punt on adding a row for FuncVersions with snapshots of each func version? just a string field for now.
+# TODO: put body and response objects have different required fields, etc; make 2 objects?
+# TODO: change all params to $ref's
+# TODO(orcl): 204 DELETE success
+# TODO(orcl): 401 - should we define this to delineate w/ 404 even tho we don't have auth by default?
+
+info:
+  title: fn
+  description: The open source serverless platform.
+  version: "0.3.0"
+# the domain of the service
+host: "127.0.0.1:8080"
+# array of all schemes that your API supports
+schemes:
+  - https
+  - http
+# will be prefixed to all paths
+# TODO(discuss): should we bump this?
+basePath: /v42
+consumes:
+  - application/json
+produces:
+  - application/json
+paths:
+  /ns:
```

Comment:
really excited to add namespacing. 👍 But do we need to make all endpoint now have the extra `/ns/` in front of their namespace and not just do `/{namespace}/apps/`. A little nitpicky i know but the `/ns/` seems wasteful. And then put namespace management under a different prefix.. Not sure where but just `/ns/` 👎  for me.  Since namespaces too me seem like a more permanent fixture and will not move around much.  But maybe thats just me.

Reply:
Will the fn server have extensions to be able to manage what happens on namespace creation deletion etc...? Seems like it should... Since that will affect things like auth for a FAAS that is using fn.. At least I feel like it probably will.

Reply:
yes, there will be extensions on CRUD for every API action, can clarify this, thanks. 

all of our resources specify which resource id follows it, e.g. `/functions/{function}`, the suggestion there would make namespace kind of an odd one out. This decision doesn't seem very consequential, we just need to be consistent I think, most people interface with the API through bindings, CLI and UI anyway so vanity matters a little less at this level; e.g. the UI can still have pretty URLs without `/thing/:thing` if we want that.

Reply:
additionally, we need a way for users to list namespaces (most implementations will attenuate this with some authentication layer) and putting this at `/ns` is the RESTful, natural way to do this.